### ISSUE TITLE
Nil pointer dereference in the btcutil package Tx.HasWitness() method

### ISSUE THREAD

**Issue by nostdm** [btcsuite/btcd#1543]
There seems to be a typo here where it checks `txHashWitness` instead of `txHasWitness` for a nil value.

Package: `btcutil`
File: `tx.go`
Line: `73`

```
func (t *Tx) HasWitness() bool {
	if t.txHashWitness != nil {
		return *t.txHasWitness
	}

	hasWitness := t.msgTx.HasWitness()
	t.txHasWitness = &hasWitness
	return hasWitness
}
```

**Comment by zkvm** [comment]
I think it's a tiny bug and i have submit a PR(https://github.com/btcsuite/btcutil/pull/156) in `btcutil` package

**Comment by onyb** [comment]
This can be closed; fixed by btcsuite/btcutil#165.
